### PR TITLE
Update Rust toolchain to 1.71.0 and remove pydocstyle workaround

### PR DESCRIPTION
**Pull request by esteve**

### CODE REVIEW

**Comment by esteve** [comment]
@maspe36 I used the workaround I suggested in https://github.com/ros2-rust/ros2_rust/issues/320#issuecomment-1633977366 and it seems to build fine locally. Let me know if you prefer the changes in PR to be split into separate PRs, I think it'd be cleaner, but given that I was already fixing `clippy` warnings I thought of putting adding the fixes all together.

**Comment by esteve** [comment]
@maspe36 never mind, I've created #322 that only addresses #320 and disables the `clippy` warning

File: rclrs/src/vendor/rcl_interfaces/msg.rs [hunk 1]
```diff
@@ -404,6 +404,7 @@ pub mod rmw {
         pub description: rosidl_runtime_rs::String,
         pub additional_constraints: rosidl_runtime_rs::String,
         pub read_only: bool,
+        pub dynamic_typing: bool,
```
Comment (maspe36) [inline rclrs/src/vendor/rcl_interfaces/msg.rs:407]: Does this break backwards compatibility with humble?

**Review by maspe36 — CHANGES_REQUESTED** [review]
The [Dockerfile](https://github.com/ros2-rust/ros2_rust/blob/main/Dockerfile#L15) also has a reference to rust 1.63 that should probably be upgraded to 1.70

**Comment by esteve** [comment]
> The [Dockerfile](https://github.com/ros2-rust/ros2_rust/blob/main/Dockerfile#L15) also has a reference to rust 1.63 that should probably be upgraded to 1.70

Done. I updated it to 1.71.0, which was released yesterday.

File: rclrs/src/vendor/rcl_interfaces/msg.rs (hunk 1 above)
Comment (esteve) [reply rclrs/src/vendor/rcl_interfaces/msg.rs:407]: Humble does have a `dynamic_typing` field as well (https://github.com/ros2/rcl_interfaces/blob/humble/rcl_interfaces/msg/ParameterDescriptor.msg#L25), so this should be ok.

**Review by maspe36 — APPROVED** [review]

**Comment by esteve** [comment]
@maspe36 thanks for reviewing these changes :slightly_smiling_face:

**Pull request merged**